buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixtureExtension.java

```diff
@@ -18,20 +18,63 @@
  */
 package org.elasticsearch.gradle.testfixtures;
 
+import org.gradle.api.GradleException;
 import org.gradle.api.NamedDomainObjectContainer;
 import org.gradle.api.Project;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 public class TestFixtureExtension {
 
     private final Project project;
     final NamedDomainObjectContainer<Project> fixtures;
+    final Map<String, String> serviceUseByProject = new HashMap<>();
 
     public TestFixtureExtension(Project project) {
         this.project = project;
         this.fixtures = project.container(Project.class);
     }
 
+    public void useFixture() {
```

Comment:
Can you explain this more? This is for cases where the fixture project uses the fixture itself, presumably for testing?

How does this work in practice? Wouldn't this mean an exclusive lock on all services defined for that fixture? How could any other project use this fixture?

Reply:
There are 3 setups in use. 

This is for when  we have a test using some fixtures and it's all contained within a project. Examples are [plugins/repository-s3/build.gradle] and [distributions/docker/build.gradle].  
So this is just a way to avoid hard coding a projects path in it's build script. 

For completeness here are the other setups: 
 
In other cases we have a fixture project and a single  other project using it. 
Some times the fixture is generic and might be used more broadly, 
others it's cleaner to have it separately. 
In these cases one doesn't have to be explicit about service names,
and will get an error if a second project tries to use that fixtures. 

Shared fixtures must be used by explicitly stating the service name.

Reply:
Got it, so to confirm, in the case where the same project both declares and uses a fixture via `useFixture()` no other projects also use that fixture?

Reply:
That's correct.